### PR TITLE
feat: add repository

### PR DESCRIPTION
**Pull request by GabyGV-GGV**
## Related Tickets & US
- [cs-56](https://app.clickup.com/t/868b3tb7a)

## Description
Allow project rules to be saved to the database using a POST endpoint. This requires updates to the data model, and related services.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
executing the tests
using swagger ui, with the provided endpoint POST /rules/{projectId}

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosApplication/36)

### CODE REVIEW

File: src/main/java/com/softdevsix/domain/repositories/RulesMemoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                .requiredCodeRating(Rating.A)
+                .build();
+
+        createRule(rule1);
+        createRule(rule2);
+        createRule(rule3);
+    }
+
+    @Override
+    public ProjectParams updateRule(ProjectParams rule) {
```
Comment (sebastian-barra-jalafund) [inline src/main/java/com/softdevsix/domain/repositories/RulesMemoryRepository.java:50]: en este metodo en lugar de devolver null es recomendable que lance lanza una excepción apropiada

**Review by sebastian-barra-jalafund — APPROVED** [review]
Looks good.

File: src/main/java/com/softdevsix/application/services/File/FileService.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.softdevsix.application.services;
+package com.softdevsix.application.services.File;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/File/FileService.java]: Directory names should be written in lowercase. Please update it.

File: src/main/java/com/softdevsix/application/services/File/IFileService.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package com.softdevsix.application.services;
+package com.softdevsix.application.services.File;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/File/IFileService.java]: Same here.

File: src/main/java/com/softdevsix/application/services/Project/IProjectService.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.softdevsix.application.services;
+package com.softdevsix.application.services.Project;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/Project/IProjectService.java]: Same here about directory name.

File: src/main/java/com/softdevsix/application/services/Project/ProjectService.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.softdevsix.application.services;
+package com.softdevsix.application.services.Project;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/Project/ProjectService.java]: Same here.

File: src/main/java/com/softdevsix/application/services/Rules/IRulesService.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.softdevsix.application.services.Rules;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/Rules/IRulesService.java]: Same.

File: src/main/java/com/softdevsix/application/services/Rules/RulesService.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import com.softdevsix.domain.entities.project.Project;
+import com.softdevsix.domain.entities.project.ProjectParams;
+import com.softdevsix.domain.exceptions.BadRequestException;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class RulesService implements IRulesService {
+    private final IProjectService PROJECT_SERVICE;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/Rules/RulesService.java]: Why do you write as a constant the variable name?

File: src/main/java/com/softdevsix/domain/repositories/IProjectRepository.java [hunk 8]
```diff
@@ -9,9 +10,10 @@
 
 @Repository
 public interface IProjectRepository {
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/domain/repositories/IProjectRepository.java:12]: The repository should extend from `JpaRepository` and its generic types, so we won't have to implement some methods from zero.

File: src/main/java/com/softdevsix/domain/repositories/IRulesRepository.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package com.softdevsix.domain.repositories;
+
+import com.softdevsix.domain.entities.project.ProjectParams;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface IRulesRepository {
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/domain/repositories/IRulesRepository.java:8]: Same here about using `JpaRepository` and its generic types.

File: src/main/java/com/softdevsix/domain/repositories/ProjectRepository.java [hunk 10]
```diff
@@ -2,29 +2,48 @@
...
 import com.softdevsix.domain.entities.project.Project;
 import org.springframework.stereotype.Repository;
-
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 @Repository
 public class ProjectRepository implements IProjectRepository {
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/domain/repositories/ProjectRepository.java:11]: When you use `JpaRepository` interface, you'll see we won't need a concrete for this class and others one.

File: src/main/java/com/softdevsix/application/services/Rules/RulesService.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        if(projectParams == null) throw new BadRequestException("Project params cannot be null");
+
+        Project project = PROJECT_SERVICE.getProjectById(projectId);
+
+        project.setProjectParams(projectParams);
+    }
+
+    @Override
+    public void executeProject() {
+
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/application/services/Rules/RulesService.java]: Why is this method empty? It should be implemented later?

File: src/main/java/com/softdevsix/domain/repositories/RulesMemoryRepository.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
+package com.softdevsix.domain.repositories;
+
+import com.softdevsix.domain.entities.project.ProjectParams;
+import com.softdevsix.domain.entities.staticanalysis.Rating;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class RulesMemoryRepository implements IRulesRepository {
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/domain/repositories/RulesMemoryRepository.java:10]: If we will use DB, why this implementation is being working?

File: src/main/java/com/softdevsix/presentation/controllers/RulesController.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import com.softdevsix.domain.exceptions.ProjectNotFoundException;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/rules/")
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/presentation/controllers/RulesController.java]: The path should follow the API conventions? Or why is like that?

File: src/main/java/com/softdevsix/presentation/controllers/RulesController.java [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+public class RulesController {
+    private final IRulesService RULES_SERVICE;
```
Comment (teran-joseluis) [inline src/main/java/com/softdevsix/presentation/controllers/RulesController.java]: Why is the name variable like a constant?

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

**Comment by teran-joseluis** [comment]
@zamoranoalicia I was talking about the comments I left with @gabygabylagaby and she explained me that those will change or will be updated later due to they were coming from another branch. I mean in the main branch and PR.

**Review by teran-joseluis — APPROVED** [review]

**Pull request merged**